### PR TITLE
BUG: Resolve CTest build errors on Linux

### PR DESCRIPTION
**Pull request by tbirdso**
Build results: https://open.cdash.org/viewBuildError.php?buildid=7744631

Resolves build failures in order for tests to build

### CODE REVIEW

File: include/itkSpectra1DAveragingImageFilter.h [hunk 1]
```diff
@@ -88,21 +88,21 @@ class ITK_TEMPLATE_EXPORT Spectra1DAveragingImageFilter : public ImageToImageFil
 
   // sets the vector length if output is a VectorImage
   template <typename TIn, typename TOut>
-  std::enable_if_t<!std::is_same_v<OutputImageType, VectorImage<OutputScalarType, TOut::ImageDimension>>>
+  std::enable_if_t<!std::is_same<OutputImageType, VectorImage<OutputScalarType, TOut::ImageDimension>>::value>
```
Comment (dzenanz) [inline include/itkSpectra1DAveragingImageFilter.h:91]: I thought `is_same_v` was available with C++14. I guess I confused it with `enable_if_t`.

File: test/itkSpectra1DAveragingImageFilterTest.cxx [hunk 2]
```diff
@@ -42,8 +42,11 @@ compareBuffers(TInImage1 * baseline, TInImage2 * test)
   itk::IdentifierType differentPixels = 0;
   itk::IdentifierType linePixels = baseline->GetLargestPossibleRegion().GetSize()[0];
   itk::IdentifierType lineCount = pixelCount / linePixels;
-  auto *              bufferBaseline = static_cast<TInImage1::PixelType *>(baseline->GetBufferPointer());
-  auto *              bufferTest = static_cast<TInImage2::PixelType *>(test->GetBufferPointer());
+
+  using InputPixelType1 = typename TInImage1::PixelType;
+  using InputPixelType2 = typename TInImage2::PixelType;
+  auto *              bufferBaseline = static_cast<InputPixelType1 *>(baseline->GetBufferPointer());
```
Comment (dzenanz) [inline test/itkSpectra1DAveragingImageFilterTest.cxx]: This now seems misaligned.

File: include/itkSpectra1DAveragingImageFilter.hxx [hunk 3]
```diff
@@ -173,7 +175,7 @@ Spectra1DAveragingImageFilter<TInputImage, TOutputImage>::GenerateData()
   // divide by the number of contributing lines
   using DividerType =
     itk::DivideImageFilter<OutputImageType, itk::Image<IdentifierType, OutputImageDimension>, OutputImageType>;
-  DividerType::Pointer divider = DividerType::New();
+  typename DividerType::Pointer divider = DividerType::New();
```
Comment (dzenanz) [inline include/itkSpectra1DAveragingImageFilter.hxx:178]: Visual Studio does not complain about missing `typename` in this situation. I always have to add it when I try compiling on Linux 😄

File: test/itkSpectra1DNormalizeImageFilterTest.cxx [hunk 4]
```diff
@@ -42,8 +42,11 @@ compareBuffers(TInImage1 * baseline, TInImage2 * test)
   itk::IdentifierType differentPixels = 0;
   itk::IdentifierType linePixels = baseline->GetLargestPossibleRegion().GetSize()[0];
   itk::IdentifierType lineCount = pixelCount / linePixels;
-  auto *              bufferBaseline = static_cast<TInImage1::PixelType *>(baseline->GetBufferPointer());
-  auto *              bufferTest = static_cast<TInImage2::PixelType *>(test->GetBufferPointer());
+
+  using InputPixelType1 = typename TInImage1::PixelType;
+  using InputPixelType2 = typename TInImage2::PixelType;
+  auto *              bufferBaseline = static_cast<InputPixelType1 *>(baseline->GetBufferPointer());
```
Comment (dzenanz) [inline test/itkSpectra1DNormalizeImageFilterTest.cxx]: Alignment. I thought that `clang-format` is run automatically as a pre-commit hook, even if your code editor doesn't invoke it.

**Review by dzenanz — COMMENTED** [review]
Mostly looks good.

File: include/itkSpectra1DAveragingImageFilter.h (hunk 1 above)
Comment (tbirdso) [reply include/itkSpectra1DAveragingImageFilter.h:91]: I had to look it up as well, looks like it is [introduced in C++17](https://en.cppreference.com/w/cpp/types/is_same)

File: test/itkSpectra1DNormalizeImageFilterTest.cxx (hunk 4 above)
Comment (tbirdso) [reply test/itkSpectra1DNormalizeImageFilterTest.cxx]: 🤔 Agreed, I guess the pre-commit hook is not running on my Linux machine for some reason. Ran reformatting back in Visual Studio and it is fixed now.

File: test/itkSpectra1DAveragingImageFilterTest.cxx (hunk 2 above)
Comment (tbirdso) [reply test/itkSpectra1DAveragingImageFilterTest.cxx]: Fixed

File: include/itkSpectra1DAveragingImageFilter.hxx (hunk 3 above)
Comment (tbirdso) [reply include/itkSpectra1DAveragingImageFilter.hxx:178]: Agreed, would be nice if MSVC went ahead and warned us on these.

**Pull request merged**